### ISSUE TITLE
Add support to pathlib.Path

### ISSUE THREAD

**Issue by turicas** [rk700/PyMuPDF#223]
A `NameError` is raised if I use `pathlib.Path` as the filename.

Code to reproduce:

```python
import pathlib
import fitz
filename = pathlib.Path('data/download/AC/2010-01-04.pdf')
doc = fitz.open(filename)
```

Using Python 3.7.0 I got this error:
```
~/software/pyenv/versions/3.7.0/envs/diarios-oficiais/lib/python3.7/site-packages/fitz/fitz.py in __init__(self, filename, stream, filetype, rect, width, height, fontsize)
   1042         if not filename or type(filename) == str:
   1043             pass
-> 1044         elif type(filename) == unicode:
   1045             filename = filename.encode('utf8')
   1046         else:
```

Replacing the last line with `doc = fitz.open(str(filename))` works.

**Comment by JorjMcKie** [comment]
Ok will be included in the next patch version (1.14.1).

I want to avoid importing `pathlib` (partly because it is not present in Python 2) and only transform the filename parameter with the `str()` function. Any comments on that?

BTW: Did you have a chance to try the new v1.14.0 w/r to suppression of MuPDF warnings?

**Comment by turicas** [comment]
@JorjMcKie I think your solution is fine.

**Comment by JorjMcKie** [comment]
published v1.14.1 here and on PyPI

**Comment by turicas** [comment]
Thanks!
